c7n/resources/cloudfront.py

```diff
@@ -436,3 +438,103 @@ def process_distribution(self, client, distribution):
                 "Exception trying to force ssl on Distribution: %s error: %s",
                 distribution['ARN'], e)
             return
+
+
+@Distribution.action_registry.register('update-distribution')
+class DistributionUpdateAction(BaseAction):
+    """Action to update the attributes of a distribution
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+        - name: enforce-distribution-logging
+          resource: distribution
+          filters:
+            - type: value
+              key: "Logging.Enabled"
+              value: null
+          actions:
+            - type: update-distribution
+              update:
+                Comment: ""
+                Enabled: true
+                Logging:
+                    Enabled: true
+                    IncludeCookies: false
+                    Bucket: 'test-enable-logging-c7n.s3.amazonaws.com'
+                    Prefix: ''
+    """
+    schema = type_schema('update-distribution',
+                        update={"type": "object"},
+                        required=('update',))
+
+    permissions = ("cloudfront:UpdateDistribution",
+                   "cloudfront:GetDistributionConfig",)
+    shape = 'UpdateDistributionRequest'
+
+    def validate(self):
+        if not self.data.get('update'):
+            raise PolicyValidationError('update parameters are missing')
+        attrs = dict(self.data.get('update'))
+        if attrs.get('CallerReference') or \
+           attrs.get('Origins') or \
+           attrs.get('DefaultCacheBehavior'):
+            raise PolicyValidationError('Invalid attribute parameter')
+
+        # Set default values for required fields
+        attrs["CallerReference"] = ""
```

Comment:
this construction would read easier by just constructing the this dict in a single statement as default and then updating attr with it.

Reply:
I've moved this construction to its own function, but since we're allowing Origins and DefaultCacheBehavior to be set now, I have to check if they exist first before overwriting them with default values.

Reply:
you don't overwrite them with defaults, you use a template dict and let the explicit values from the config update the defaults.